benches/lib.rs

```diff
@@ -33,8 +32,30 @@ mod to_html {
         &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA;
         &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA;
         &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA; &#xAAA;";
-        let mut opts = Options::empty();
 
-        b.iter(|| render_html(&input, opts));
+        b.iter(|| render_html(&input, Options::empty()));
     }
-}
\ No newline at end of file
+
+    #[bench]
+    fn pathological_links(b: &mut test::Bencher) {
+        let input = std::iter::repeat("[a](<").take(1000).collect::<String>();
+
+        b.iter(|| render_html(&input, Options::empty()));
+    }
+
+    #[bench]
+    fn pathological_codeblocks(b: &mut test::Bencher) {
```

Comment:
This is not addressed directly in this PR, but maybe is one of the next things to work on?

Reply:
I argue in the final commit message that we don't need to: https://github.com/raphlinus/pulldown-cmark/pull/237/commits/f756258c8fdd438451df94646f8dce0617768449

Reply:
Ah. There's still super-linear growth if your input is the code block sequence (O(n) AST nodes, O(n^2) bytes), followed by a sequence of O(n^2) AST nodes (i.e. `"*a* "` repeated n^2 times). Then time is O(n^3), so O(n^1.5) relative to input size.

Reply:
I believe that you are correct. Nice example! Then indeed we will need to do some extra work :-)

What would be a good way of dealing with this? We could keep a list of non-matching code blocks delimiters whenever we are scanning for a matching one. If it fails, it will contain all code block delimiters of the paragraph. Subsequent code block scans in the same paragraph could scan that vector instead of the AST. I think it should guard against your example.